File: packages/sdkwork-utils-csharp/BooleanUtils.cs
namespace Sdkwork.Utils;

public static class BooleanUtils
{
    public static bool? ParseBool(string value)
    {
        return value.Trim().ToLowerInvariant() switch
        {
            "true" or "1" or "yes" or "on" => true,
            "false" or "0" or "no" or "off" => false,
            _ => null,
        };
    }

    public static bool IsTruthy(string? value)
    {
        if (string.IsNullOrWhiteSpace(value)) return false;
        return value.Trim().ToLowerInvariant() switch
        {
            "false" or "0" or "no" or "off" => false,
            _ => true,
        };
    }
}

File: packages/sdkwork-utils-php/src/BooleanUtils.php
<?php

declare(strict_types=1);

namespace Sdkwork\Utils;

final class BooleanUtils
{
    public static function parseBool(string $value): ?bool
    {
        return match (strtolower(trim($value))) {
            'true', '1', 'yes', 'on' => true,
            'false', '0', 'no', 'off' => false,
            default => null,
        };
    }

    public static function isTruthy(?string $value): bool
    {
        if ($value === null || trim($value) === '') {
            return false;
        }

        return !in_array(strtolower(trim($value)), ['false', '0', 'no', 'off'], true);
    }
}
